SensorController/Core/Src/Peripherals/i2c.c

```diff
@@ -34,188 +35,41 @@ void I2C_Init(void)
+HAL_StatusTypeDef DAT_SensorIsReady(DAT_SENSOR *dev) {
 
-	return HAL_I2C_IsDeviceReady(dev->i2c_handle, (dev->LSM6DS3_ADDR << 1), 10, HAL_MAX_DELAY);
+	return HAL_I2C_IsDeviceReady(dev->I2C_HANDLE, (dev->SENSOR_ADDR << 1), 10, HAL_MAX_DELAY);
 }
 
 /* Low level HAL Function replacements for use in tasks etc. abstracts some HAL constants out */
+// TODO just use sempaphore stuff lol
 
-HAL_StatusTypeDef LSM6DS3_readRegisters(LSM6DS3 *dev,uint8_t reg, uint8_t * data, uint8_t length){
-	return HAL_I2C_Mem_Read(dev->i2c_handle,(dev->LSM6DS3_ADDR << 1), reg, I2C_MEMADD_SIZE_8BIT, data, length, HAL_MAX_DELAY);
-}
-
-
-HAL_StatusTypeDef LSM6DS3_writeRegister(LSM6DS3 *dev,uint8_t reg, uint8_t * data){
-	return	HAL_I2C_Mem_Write(dev->i2c_handle, (dev->LSM6DS3_ADDR << 1), reg, I2C_MEMADD_SIZE_8BIT, data, 1,HAL_MAX_DELAY);
-}
-
-HAL_StatusTypeDef LSM6DS3_Init(LSM6DS3 * dev, I2C_HandleTypeDef * i2cHandle) {
-
-	HAL_StatusTypeDef sensor_ret;
-	dev->i2c_handle = i2cHandle;
-	dev->LSM6DS3_ADDR =  LOCKED_ADDR;
-	sensor_ret = HAL_I2C_IsDeviceReady(dev->i2c_handle, (dev->LSM6DS3_ADDR << 1), 2, HAL_MAX_DELAY);
-	if(sensor_ret != HAL_OK) return sensor_ret;
-
-	//Example initializations of data arrays for output
-	dev->accel_data[0] = 0.0f;
-	dev->accel_data[1] = 0.0f;
-	dev->accel_data[2] = 0.0f;
-	
-	dev->temp_data = 0.0f;
-
-	return sensor_ret;
-}
-
-/* Processing Functions */
-
-HAL_StatusTypeDef LSM6DS3_ReadTemp(LSM6DS3 * dev){
-
-	//Data is 2 bytes signed two's complement
-
-	uint8_t  regData[2]; //Temp data comes in two bytes
-
-	HAL_StatusTypeDef status = LSM6DS3_readRegisters(dev, OUT_TEMP_L, regData, 2);
-
-	if (status != HAL_OK)
-	{
-		return status;
-	}
-	
-	int16_t signed_twobyte_temp = 0x0000;
-
-#ifdef TEST_CONVERSION
-	//source
-	//https://github.com/sparkfun/SparkFun_LSM6DS3_Arduino_Library/blob/master/src/SparkFunLSM6DS3.cpp#L747
-
-	signed_twobyte_temp |= ((int16_t)regData[0] |(int16_t)regData[1]<<8);
-
-	float temp_output = (float) signed_twobyte_temp /16; // divide 16 to scale
-	temp_output += 25; // add to remove offset
-	dev->temp_data = temp_output;
-#endif
-	return status;
-
-}
-
-HAL_StatusTypeDef LSM6DS3_Reg_Init(LSM6DS3 *dev, uint8_t Type) {
-
-	HAL_StatusTypeDef status;
-
-	if(Type == ACCEL_ONLY_ENABLE) {
-		uint8_t *PregConfig;
-		uint8_t regConfig = 0x4A;
-		PregConfig = &regConfig;
-		/* Configures the Accelerometer control register
-		 * Configuration: 0b01001010
-		 * Freq: 104Hz < -- Ouput Data Rate Register = 0100
-		 * Full-Scale Selction: 4g <-- FS_XL = 10
-		 * Bandwith Filter Selection: 100Hz <-- BW_XL = 10
-		 */
-		LSM6DS3_writeRegister(dev, CTRL1_XL, PregConfig);
-	}
-	else if (Type == GYRO_ONLY_ENABLE) {
-		//pg. 27
-		//write to CTRL2_G
-		//write power mode to CTRL7_G
-	}
-	else {
-		//init both
-	}
-
-	return status;
+HAL_StatusTypeDef DAT_ReadRegisters(DAT_SENSOR *dev,uint8_t reg, uint8_t * data, uint8_t length, uint8_t blockTime){
+	HAL_StatusTypeDef ret;
+	xSemaphoreTake(I2CCommandSemphr, pdMS_TO_TICKS(blockTime));
```

Comment:
This mutex should also be used in the DAT_WriteReg function

Reply:
Weird.. It is on my version
`
HAL_StatusTypeDef DAT_WriteRegister(DAT_SENSOR *dev,uint8_t reg, uint8_t * data, uint8_t length, uint8_t blockTime){
	HAL_StatusTypeDef ret;
	xSemaphoreTake(I2CCommandSemphr, pdMS_TO_TICKS(blockTime));
	ret = HAL_I2C_Mem_Write(dev->I2C_HANDLE, (dev->SENSOR_ADDR << 1), reg, I2C_MEMADD_SIZE_8BIT, data, length, HAL_MAX_DELAY);
	xSemaphoreGive(I2CCommandSemphr);
	return ret;
}`

Reply:
nvm it wasn't showing that function on the PR display, but I can see it in the file. Ur good on this!